### ISSUE TITLE
PauseOnScrollListener does not support RecyclerView

### ISSUE THREAD

**Issue by Syrou87** [nostra13/Android-Universal-Image-Loader#851]
I already made an extension which allows PauseOnScrollListener to work under the new Lollipop RecyclerView scroll, but I thought I should mention this issue, so maybe it gets added to the library itself :)

**Comment by kyze8439690** [comment]
I have wrote one as below:

    import android.support.v7.widget.RecyclerView;

    import com.nostra13.universalimageloader.core.ImageLoader;

    import static android.support.v7.widget.RecyclerView.OnScrollListener;

    public class PauseOnScrollListener2 extends OnScrollListener{

        private ImageLoader mImageLoader;

        private final boolean mPauseOnScroll;
        private final boolean mPauseOnFling;
        private final OnScrollListener mExternalListener;

        public PauseOnScrollListener2(ImageLoader imageLoader, boolean pauseOnScroll, boolean pauseOnFling) {
            this(imageLoader, pauseOnScroll, pauseOnFling, null);
        }

        public PauseOnScrollListener2(ImageLoader imageLoader, boolean pauseOnScroll, boolean pauseOnFling,
                                  OnScrollListener customListener) {
            this.mImageLoader = imageLoader;
            this.mPauseOnScroll = pauseOnScroll;
            this.mPauseOnFling = pauseOnFling;
            mExternalListener = customListener;
        }

        @Override
        public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
            if (mExternalListener != null) {
                mExternalListener.onScrolled(recyclerView, dx, dy);
            }
        }

        @Override
        public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
            switch (newState) {
                case RecyclerView.SCROLL_STATE_IDLE:
                    mImageLoader.resume();
                    break;
                case RecyclerView.SCROLL_STATE_DRAGGING:
                    if (mPauseOnScroll) {
                        mImageLoader.pause();
                    }
                    break;
                case RecyclerView.SCROLL_STATE_SETTLING:
                    if (mPauseOnFling) {
                        mImageLoader.pause();
                    }
                    break;
            }
            if (mExternalListener != null) {
                mExternalListener.onScrollStateChanged(recyclerView, newState);
            }
        }
    }

It works, but the only problem is that I need to add RecyclerView dependence, which I think not necessary for UIL.

**Comment by Syrou87** [comment]
Yeah Mine is pretty much the same but used another naming convention: 

package se.appcorn.androidbaseproject.listeners;

import android.support.v7.widget.RecyclerView;
import android.widget.AbsListView;

import com.nostra13.universalimageloader.core.ImageLoader;

/**
 * Created by jfo on 18/12/14.
 * Adds PauseOnScrollListener support to Universal Image Loader for ReyclerView
 */
public class RecyclerPauseOnScrollListener extends RecyclerView.OnScrollListener {

    private ImageLoader imageLoader;

    private final boolean pauseOnScroll;
    private final boolean pauseOnFling;
    private final RecyclerView.OnScrollListener externalListener;

    /**
     * Constructor
     *
     * @param imageLoader   {@linkplain ImageLoader} instance for controlling
     * @param pauseOnScroll Whether {@linkplain ImageLoader#pause() pause ImageLoader} during touch scrolling
     * @param pauseOnFling  Whether {@linkplain ImageLoader#pause() pause ImageLoader} during fling
     */
    public RecyclerPauseOnScrollListener(ImageLoader imageLoader, boolean pauseOnScroll, boolean pauseOnFling) {
        this(imageLoader, pauseOnScroll, pauseOnFling, null);
    }

    /**
     * Constructor
     *
     * @param imageLoader    {@linkplain ImageLoader} instance for controlling
     * @param pauseOnScroll  Whether {@linkplain ImageLoader#pause() pause ImageLoader} during touch scrolling
     * @param pauseOnFling   Whether {@linkplain ImageLoader#pause() pause ImageLoader} during fling
     * @param customListener Your custom {@link android.widget.AbsListView.OnScrollListener} for {@linkplain AbsListView list view} which also will
     *                       be get scroll events
     */
    public RecyclerPauseOnScrollListener(ImageLoader imageLoader, boolean pauseOnScroll, boolean pauseOnFling, RecyclerView.OnScrollListener customListener) {
        this.imageLoader = imageLoader;
        this.pauseOnScroll = pauseOnScroll;
        this.pauseOnFling = pauseOnFling;
        externalListener = customListener;
    }

    @Override
    public void onScrollStateChanged(RecyclerView view, int scrollState) {
        switch (scrollState) {
            case AbsListView.OnScrollListener.SCROLL_STATE_IDLE:
                imageLoader.resume();
                break;
            case AbsListView.OnScrollListener.SCROLL_STATE_TOUCH_SCROLL:
                if (pauseOnScroll) {
                    imageLoader.pause();
                }
                break;
            case AbsListView.OnScrollListener.SCROLL_STATE_FLING:
                if (pauseOnFling) {
                    imageLoader.pause();
                }
                break;
        }
        if (externalListener != null) {
            externalListener.onScrollStateChanged(view, scrollState);
        }
    }

    @Override
    public void onScrolled(RecyclerView view, int dx, int dy) {
        if (externalListener != null) {
            externalListener.onScrolled(view, dx, dy);
        }
    }

}
